ksql-parser/src/main/java/io/confluent/ksql/parser/tree/InsertValues.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.parser.tree;
+
+import com.google.common.collect.ImmutableList;
+import com.google.errorprone.annotations.Immutable;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Immutable
+public class InsertValues extends Statement {
+
+  private final QualifiedName target;
+  private final ImmutableList<String> columns;
+  private final ImmutableList<Expression> values;
+
+  public InsertValues(
+      final QualifiedName target,
+      final List<String> columns,
+      final List<Expression> values
+  ) {
+    this(Optional.empty(), target, columns, values);
+  }
+
+  public InsertValues(
+      final Optional<NodeLocation> location,
+      final QualifiedName target,
+      final List<String> columns,
+      final List<Expression> values
+  ) {
+    super(location);
+    this.target = Objects.requireNonNull(target, "target");
+    this.columns = ImmutableList.copyOf(Objects.requireNonNull(columns, "columns"));
+    this.values = ImmutableList.copyOf(Objects.requireNonNull(values, "values"));
+
+    if (!columns.isEmpty() && columns.size() != values.size()) {
+      throw new IllegalArgumentException(
```

Comment:
We should throw KsqlException in this case since indicates a user error. The rest layer throws 400 when it catches KsqlException, and 500 for other runtime exceptions.